partiql-value/src/time.rs

```diff
@@ -1,191 +1,138 @@
+            (None, Some(m)) => Some(UtcOffset::from_hms(0, m, 0).expect("valid offset")),
+            (Some(h), None) => Some(UtcOffset::from_hms(h, 0, 0).expect("valid offset")),
+            _ => None,
+        };
+
+        Self::from_hmfs_offset(hour, minute, second, offset)
+    }
+
+    pub fn from_ymd(year: i32, month: NonZeroU8, day: u8) -> Self {
+        let month: time::Month = month.get().try_into().expect("valid month");
+        let date = time::Date::from_calendar_date(year, month, day).expect("valid ymd");
+        TimeValue::Date(date)
+    }
+
+    pub fn from_ymdhms(
+        year: i32,
+        month: NonZeroU8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: f64,
+    ) -> Self {
+        let month: time::Month = month.get().try_into().expect("valid month");
+        let date = time::Date::from_calendar_date(year, month, day).expect("valid ymd");
+        let time = time_from_hmfs(hour, minute, second);
+        let date = date.with_time(time);
+        TimeValue::DateTime(date)
+    }
+
+    fn from_hmfs_offset(hour: u8, minute: u8, second: f64, offset: Option<UtcOffset>) -> Self {
+        let time = time_from_hmfs(hour, minute, second);
+        match offset {
+            Some(offset) => TimeValue::TimeWithTz(time, offset),
+            None => TimeValue::Time(time),
+        }
     }
 }
 
-impl IntoIterator for Bag {
-    type Item = Value;
-    type IntoIter = BagIntoIterator;
-
-    fn into_iter(self) -> BagIntoIterator {
-        BagIntoIterator(self.0.into_iter())
-    }
+fn time_from_hmfs(hour: u8, minute: u8, second: f64) -> time::Time {
+    let millis = (second.fract() * 1e9) as u32;
+    let second = second.trunc() as u8;
+    time::Time::from_hms_nano(hour, minute, second, millis).expect("valid time value")
 }
 
-pub struct BagIntoIterator(vec::IntoIter<Value>);
-
-impl Iterator for BagIntoIterator {
-    type Item = Value;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        self.0.next()
-    }
-}
-
-impl Debug for Bag {
+impl Debug for TimeValue {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "<<")?;
-        let mut iter = self.iter().peekable();
-        while let Some(v) = iter.next() {
-            if iter.peek().is_some() {
-                write!(f, "{v:?}, ")?;
-            } else {
-                write!(f, "{v:?}")?;
+        match self {
+            TimeValue::Date(d) => {
+                write!(f, "DATE '{:?}'", d)
+            }
+            TimeValue::Time(t) => {
+                write!(f, "TIME '{:?}'", t)
+            }
+            TimeValue::TimeWithTz(t, tz) => {
+                write!(f, "TIME WITH TIME ZONE '{:?} {:?}'", t, tz)
+            }
+            TimeValue::DateTime(dt) => {
+                write!(f, "TIMESTAMP '{:?}'", dt)
+            }
+            TimeValue::DateTimeWithTz(dt) => {
+                write!(f, "TIMESTAMP WITH TIME ZONE '{:?}'", dt)
```

Comment:
As a side note, even though the SQL92 spec defines timestamps this way, I don't think the spec does this formatting:

Found [page 8 figure 2](https://partiql.org/assets/PartiQL-Specification.pdf#page=8) writes a timestamp with the Ion value backticks:
```
{
    ...
    ‘configurationItemCaptureTime’: `2016-08-03T08:56:52.415Z`,
    ...
}
```
while later in the spec [section 11.3](https://partiql.org/assets/PartiQL-Specification.pdf#subsection.11.3) it includes a “timestamp” (really is a time value)
```
logs: [{‘sensor’:1, ‘co’:0.4, ‘timestamp’:04:05:06},
       {‘sensor’:1, ‘co’:0.2, ‘timestamp’:04:05:07},
       {‘sensor’:1, ‘co’:0.5, ‘timestamp’:04:05:10},
       {‘sensor’:2, ‘co’:0.3}
]
```

Also the Kotlin reference implementation doesn't support timestamps entered using the SQL92 syntax. And functions outputting timestamps output using the Ion value backticks.

Perhaps we can cut issues to `partiql-lang-kotlin` and/or `partiql-spec` to support the SQL-92 syntax.